Assets/Scripts/UI/Pickup Pup/ShelterScreen/PPShelterUIController.cs

```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * Author: Isaiah Mann
+ * Description: Controls the shelter screen
+ */
+
+public class PPShelterUIController : PPUIController
+{
+    DogSlot[] availableDogPortraits;
+    DogDatabase database;
+
+    #region MonoBehaviourExtended Overrides
+
+    protected override void setReferences()
+    {
+        base.setReferences();
+        availableDogPortraits = GetComponentsInChildren<DogSlot>();
+    }
+
+    protected override void fetchReferences()
+    {
+        base.fetchReferences();
+        database = DogDatabase.GetInstance;
+        populateAvailableDogs(database);
+    }
+
+    #endregion
+
+    void populateAvailableDogs(DogDatabase database)
```

Comment:
Confirming we want a new random selection each time the script runs? Not a random selection every realtime day or something more complex